domain-mapping/api/class-darkmatter-domains.php

```diff
@@ -111,6 +111,46 @@ private function _basic_check( $fqdn = '' ) {
 		return apply_filters( 'darkmatter_domain_basic_check', $fqdn );
 	}
 
+	/**
+	 * Clears out the relevant caches, usually when a domain is added / updated / deleted.
+	 *
+	 * @since 2.2.0
+	 *
+	 * @param string $fqdn Fully qualified domain name. Optional.
+	 */
+	private function _clear_cache( $fqdn = '' ) {
```

Comment:
I would create 3 `get_cache_key_x()` methods so that the `_clear_cache` responsibility is clearing the cache not setting up the cache keys. The current code is confusing as the `cache_key` variable is overwritten multiple times and that would be solved as well.

```suggestion
	private function _clear_caches( $fqdn = '' ) {
```

Reply:
Excellent suggestion and yes, I've refactored into two methods - one for clearing the cache for the specific domain and another for clearing the cache for the domain types (which is reusable for "main" and "media" domains).

Also made the clear cache methods public as I can theorise - although not quite imagine... - a scenario where custom logic made use some of the hooks and then need to clear the cache out based on their own criteria.